File: portal-backend/src/main/java/com/empresa/portal_backend/dto/ProjectRequest.java
package com.empresa.portal_backend.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

import java.util.List;

/**
 * Solicitud para crear o actualizar un proyecto.
 */
public record ProjectRequest(
        /** Nombre del proyecto */
        @NotBlank String name,
        /** Descripción detallada del proyecto */
        String description,
        /** Lista de tecnologías utilizadas (tags) */
        @NotNull List<String> tags,
        /** Estado actual del proyecto */
        @NotBlank String status
) {
}

File: portal-frontend/src/app/models/project.request.ts
/**
 * Datos para crear o actualizar un proyecto (coincide con ProjectRequest del backend).
 */
import { Project } from './project.model';

export interface ProjectRequest {
  /** Nombre del proyecto */
  name: string;
  /** Descripción detallada */
  description: string;
  /** Tecnologías utilizadas (tags) */
  tags: string[];
  /** Estado actual del proyecto */
  status: Project['status'];
}
